### DISCUSSION TITLE
SQL中字符串能否添加双引号

### DISCUSSION THREAD

**Discussion by ghost** [liusy58/NJU_AdvancedProgramming_SP22#89] [Q&A] [answered]
如题，如果所有SQL语句中字符串都加了双引号，算不算 **“修改基本需求”** ？
如果字符串没有双引号，在完全合法的输入下，一些插入语句和查询语句会产生 **二义性**
例如商品description可能包含英文逗号，如果没有双引号，insert和update语句都可能产生二义性。
如果不允许加双引号，也可以采用某种约定好的编码规避掉这个问题，但是commands.txt的输出内容又不符合要求了

**Comment by ghost** [comment]
换言之，当输入数据会造成SQL问题时，是否可以生成非标准的语句？
当输入数据为'aaa'时，对应语句 insert into ... values (1,2,aaa,4)
当输入数据为'a,a'时，原语法一定会出问题，那我是否能生成insert into ... values (1,2,%612c61,4)之类的，自定义的语法？

**Reply by ghost** [reply]
![image](https://user-images.githubusercontent.com/100082486/156167083-5d34fd59-83e3-42f9-922d-7ba0a0fbc943.png)

**Comment by RicoloveFeng** [comment, marked as answer]
我倾向于约定描述里面不包含逗号，当然加双引号也是一种办法，但是可能会增加工作量。具体看实验设计助教怎么说，我们应该不会针对这一点做测试用例。

**Reply by ghost** [reply]
如果无脑给所有字符串加引号，那么command.txt的输出必然和标准输出不同。
如果采用 发现**可能**造成危险的字符就编码的方案，输出在一定情况下也和标准输出不同。
在测试时如果发现这种情况，会不会扣分呢？

**Reply by binbin00** [reply]
这是我们设计的纰漏之处，现在就默认description里不带英文逗号。
